### ISSUE TITLE
override checkout submit ajax better documentation or ability to do it from gateways api

### ISSUE THREAD

**Issue by thierrydev** [woocommerce/woocommerce#26501]
**Is your feature request related to a problem? Please describe.**
I'm trying to override the checkout submit form action/Ajax when my gateway is selected as the normal WC checkout sequence is not relevant to my use case 
[checkout.js#L35](
https://github.com/woocommerce/woocommerce/blob/2c192410a27221219c1440c8b6fb9b5d0696e2c2/assets/js/frontend/checkout.js#L35)

**Describe the solution you'd like**

- add a filter to override the submit form action completely
- add better documentation of how to override the form action with jQuery

**Describe alternatives you've considered**
my only course of action is to override the submit button Ajax call.
**Additional context**

**Comment by Konamiman** [comment]
Hi @thierrydev, thanks for your enhancement suggestion.

Just to make sure that we're speaking about the same thing: there's a `woocommerce_get_checkout_url` filter that allows replacing the action attribute of the checkout form. Would this be compatible with the first solution you're suggesting?

**Comment by thierrydev** [comment]
Hello

Thanks for getting back to me.

Does that filter override the forms Ajax call entirely ?

**Comment by Konamiman** [comment]
> Does that filter override the forms Ajax call entirely ?

No, sorry, I think I misunderstood you. What this filter does is to replace the value of the `action` attribute in the `form` element, but what you need is to execute a custom JS function when the submit button is clicked, instead of "`this.submit`", right? In that case, indeed, changes are required (probably a new filter) to achieve that.

**Comment by thierrydev** [comment]
That is exactly what I'm looking for a new filter to modify the "this.submit"

thank you very much love what you guys do

**Comment by jamesckemp** [comment]
This issue has been automatically closed as part of the backlog cleanup announced in [Refining how we manage the WooCommerce backlog](https://developer.woocommerce.com/2026/05/29/backlog-cleanup/). It met the criteria outlined there. If this still affects a current version of WooCommerce, please follow the steps in that post. A fresh report with current reproduction steps is the fastest path to triage. If this is a feature request, please add it to the [WooCommerce feature request board](https://woocommerce.com/feature-requests/woocommerce/) so it can be considered alongside other requests.
